### PR TITLE
[#107171176] Convert Microsoft.rest.azure.authorization library to coreclr and adal 3.5

### PR DESCRIPTION
**Pull request by haocs**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@haocs__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by stankovski** [comment]
@haocs looks like the changes to nuspec file as well as project.json never made it to this PR. Also, please update the build.proj to run dnu build against this project.

File: ClientRuntimes/CSharp/ClientRuntime.Azure.Authentication/UserTokenProvider.cs [hunk 1]
```diff
@@ -261,33 +261,27 @@ public class UserTokenProvider : ITokenProvider
         public static async Task<ServiceClientCredentials> LoginWithPromptAsync(string domain, ActiveDirectoryClientSettings clientSettings,
```
Comment (markcowl) [inline ClientRuntimes/CSharp/ClientRuntime.Azure.Authentication/UserTokenProvider.cs]: @haocs I wonder if we're going to need to change this signature cross-platform.  I suspect we will need to expose PlatformProvider, I'm not sure if the default PlatfromProvider will work cross-platform.  We should ask the ADAL team about this.

Comment (haocs) [inline ClientRuntimes/CSharp/ClientRuntime.Azure.Authentication/UserTokenProvider.cs]: @stankovski I think this PR is for updating ADAL to v3.5. Will be another PR including the nuspec, json and dnu build changes.
